src/components/CurrencyInputPanel/index.tsx

```diff
@@ -235,9 +239,26 @@ export default function CurrencyInputPanel({
               </Text>
             )}
             {account && currency && !disabled && showMaxButton && label !== 'To' && (
-              <Button onClick={onMax} scale="xs" variant="secondary" style={{ textTransform: 'uppercase' }}>
-                {t('Max')}
-              </Button>
+              <Flex alignItems="right" justifyContent="right">
+                {showQuickInputButton &&
```

Comment:
can move out the `showMaxButton` check
we can show this component when user has balance

Reply:
`selectedCurrencyBalance?.greaterThan(0)`